### PR TITLE
Feature 107 vision common

### PR DESCRIPTION
**Pull request by chutsu**
In relation to #107 

#### Pre-Merge Checklist:
- [x] Code is documented.
- [x] Code has automated tests.
- [x] Zero compiler warnings with `-Wall -Wextra`
- [x] Formatted with `clang-format`

#### Details
Add focal length, matrix concatenation and projection matrix utilities

### CODE REVIEW

File: wave_vision/include/wave/vision/common.hpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * pixels */
+Vec2 focal_length(double hfov,
+                  double vfov,
+                  double image_width,
+                  double image_height);
+
+/** Combine 2 matrices together to form a third, this function was design to
+ * concatenate two image matrices `img1` and `img2` so they can form a third
+ * side by side image `out` */
+void matconcat(const cv::Mat &img1, const cv::Mat &img2, cv::Mat &out);
```
Comment (leokoppel) [inline wave_vision/include/wave/vision/common.hpp]: Use [cv::hconcat](http://docs.opencv.org/trunk/d2/de8/group__core__array.html#gaab5ceee39e0580f879df645a872c6bf7)

File: wave_vision/include/wave/vision/common.hpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+#include "wave/utils/utils.hpp"
+
+namespace wave {
+
+/** Calculate theoretical focal length based on field of view (fov) in radians
+ * and image width in pixels */
+double focal_length(double fov, double image_width);
+
+/** Calculate theoretical focal length based on field of view (fov) in
```
Comment (leokoppel) [inline wave_vision/include/wave/vision/common.hpp]: Doxygen uses the first sentence as a "brief" description. So generally keep the first sentence to a few words, and describe the parameters with `@param` tags (or just on the next line).

Also, inconveniently, we need the `@addtogroup vision` tags inside the `namespace wave`.

I have made these changes.

File: wave_vision/src/common.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    return (image_width / 2.0) / tan(deg2rad(hfov) / 2.0);
+}
+
+Vec2 focal_length(double hfov,
+                  double vfov,
+                  double image_width,
+                  double image_height) {
+    Vec2 focal;
+
+    focal(0) = (image_width / 2.0) / tan(deg2rad(hfov) / 2.0);
```
Comment (leokoppel) [inline wave_vision/src/common.cpp]: Could call the other `focal_length`.

File: wave_vision/include/wave/vision/common.hpp (hunk 2 above)
Comment (chutsu) [inline wave_vision/include/wave/vision/common.hpp]: Noted

File: wave_vision/include/wave/vision/common.hpp (hunk 1 above)
Comment (chutsu) [inline wave_vision/include/wave/vision/common.hpp]: removed

File: wave_vision/src/common.cpp (hunk 3 above)
Comment (chutsu) [inline wave_vision/src/common.cpp]: Done

**Pull request merged**